lib/core/src/Cardano/Wallet.hs

```diff
@@ -708,6 +713,14 @@ restoreBlocks ctx wid blocks nodeTip = db & \DBLayer{..} -> do
         <$> withNoSuchWallet wid (readCheckpoint $ PrimaryKey wid)
         <*> withNoSuchWallet wid (readWalletMeta $ PrimaryKey wid)
     let bp = blockchainParameters cp
+
+    unless (cp `isParentOf` NE.head blocks) $ fail $ T.unpack $ T.unwords
+        [ "restoreBlocks: given chain isn't a valid continuation."
+        , "Wallet is at:", pretty (currentTip cp)
+        , "but the given chain continues starting from:"
+        , pretty (header (NE.head blocks))
+        ]
```

Comment:
I added this when I noticed something off with the logs and blocks being applied. In my first attempt solving this I didn't quite look into re-setting the intersection with the node, so on restart, the workers were fetching back from genesis... and re-applying blocks as normal :| ... So, I left it there, as an invariant check.